### DISCUSSION TITLE
How to filter based on multiple many to many relationships

### DISCUSSION THREAD

**Discussion by shellking4** [pocketbase/pocketbase#3442] [Q&A] [answered]
Hey 
I have this set of relationship, a course has many tags and many levels. 
Now I want to filter courses based on tags and levels

How can I do it ? 

in sql I'd do something like: tag IN [...tags] && level IN [...levels]

**Comment by ganigeorgiev** [comment, marked as answer]
You can use dot-notation to access the specific relation field and combine the different conditions with `||` (OR) clause:

```js
tags.name ?= "someTag1" || tags.name ?= "someTag2" ...
```
_(the `?` operator prefix means "any/at-least-one" of the tags items must satisfy the condition)_

If you have the tags you want to filter as an array you can also do something like:

```js
const tags = ["someTag1", "someTag2"];

const records = await pb.collection("courses").getFullList({
    filter: tags.map((tag) => `tags.name?="${tag}"`).join("||")
})
```

**Reply by shellking4** [reply]
Thank you for this awesome answer. It worked
